### PR TITLE
DWPF-1715 Add Navigation Card component

### PR DESCRIPTION
**Pull request by nicopicchio**
- [x] JIRA ticket referenced in title
- [x] Title is clear and concise
- [x] Description gives any relevant detail
- [ ] Tests are up to date (N/A)
- [ ] Documentation is up to date (N/A)

### CODE REVIEW

File: src/dw_design_system/dwds/components/navigation_card.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+from wagtail import blocks
+
+
+class NavigationCardBlock(blocks.StructBlock):
+    """A nav card to direct users"""
+
+    title = blocks.CharBlock(required=True)
+    url = blocks.URLBlock(required=True)
```
Comment (marcelkornblum) [inline src/dw_design_system/dwds/components/navigation_card.py:8]: wondering if this should be a page chooser?

File: src/dw_design_system/dwds/components/navigation_card.scss [hunk 2]
```diff
@@ -0,0 +1,14 @@
+.dwds-navigation-card {
+    display: flex;
```
Comment (marcelkornblum) [inline src/dw_design_system/dwds/components/navigation_card.scss:2]: i'd love if we could use the new primitives for this instead, but not if it's a big pain

**Review by marcelkornblum — COMMENTED** [review]
looking good! a few bits and bobs

File: src/content/models.py [hunk 3]
```diff
@@ -434,6 +434,7 @@ class NavigationPage(BasePage):
         [
             ("dw_curated_page_links", dwds_blocks.CustomPageLinkListBlock()),
             ("dw_cta", dwds_blocks.CTABlock()),
+            ("dw_navigation_card", dwds_blocks.NavigationCardBlock()),
```
[the remark this answers is not in the document]
Comment (nicopicchio) [reply src/content/models.py:437]: Sure, I noticed you deleted them and I thought you didn't want it in there.

File: src/dw_design_system/dwds/components/navigation_card.py (hunk 1 above)
Comment (nicopicchio) [reply src/dw_design_system/dwds/components/navigation_card.py:8]: Good point. I'll change it now.

File: src/content/models.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (marcelkornblum) [reply src/content/models.py:437]: yeah this page should only have nav cards in the primary, and all the nav elements in the secondary

**Review by alsyx — APPROVED** [review]

**Pull request merged**